File: samples/conference/registration-bc/src/main/java/com/microsoft/conference/registration/readmodel/QueryServices/Implementation/PaymentQueryService.cs
using System;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using Conference.Common;
using ECommon.Components;
using ECommon.Dapper;

namespace Registration.ReadModel.Implementation
{
    [Component]
    public class PaymentQueryService : IPaymentQueryService
    {
        public Payment FindPayment(Guid paymentId)
        {
            using (var connection = GetConnection())
            {
                return connection.QueryList<Payment>(new { Id = paymentId }, ConfigSettings.PaymentTable).FirstOrDefault();
            }
        }

        private IDbConnection GetConnection()
        {
            return new SqlConnection(ConfigSettings.ConferenceConnectionString);
        }
    }
}

File: samples/conference/registration-bc/src/main/java/com/microsoft/conference/registration/readmodel/QueryServices/Implementation/PaymentQueryService.java
package com.microsoft.conference.registration.readmodel.QueryServices.Implementation;

import com.microsoft.conference.registration.readmodel.QueryServices.IPaymentQueryService;
import com.microsoft.conference.registration.readmodel.QueryServices.Payment;

public class PaymentQueryService implements IPaymentQueryService {
    @Override
    public Payment FindPayment(String paymentId) {
        return null;
    }
//        public Payment FindPayment(String paymentId)
//        {
//            using (var connection = GetConnection())
//            {
//                return connection.QueryList<Payment>(new { Id = paymentId }, ConfigSettings.PaymentTable).FirstOrDefault();
//            }
//        }
//
//        private IDbConnection GetConnection()
//        {
//            return new SqlConnection(ConfigSettings.ConferenceConnectionString);
//        }
}
